Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Meter/MeterMaintenance.tsx

```diff
@@ -0,0 +1,257 @@
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  01/16/2020 - Billy Ernest
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Application, OpenXDA } from '@gpa-gemstone/application-typings';
+import Table from '@gpa-gemstone/react-table';
+import { Pencil, TrashCan } from '@gpa-gemstone/gpa-symbols';
+import { Warning, Modal, LoadingScreen, ServerErrorIcon } from '@gpa-gemstone/react-interactive';
+import { CheckBox, DatePicker } from '@gpa-gemstone/react-forms';
+
+declare var homePath: string;
+
+interface IProps { Meter: OpenXDA.Types.Meter }
+
+//TODO: Move to Gemstone when updating application typings
+interface MaintenanceWindow {
+    ID: number,
+    MeterID: number,
+    StartTime: string,
+    EndTime: string
+}
+
+const MeterMaintenanceWindow = (props: IProps) => {
+    // Table Consts
+    const [sortKey, setSortKey] = React.useState<keyof MaintenanceWindow>('StartTime');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+    const [status, setStatus] = React.useState<Application.Types.Status>('unintiated');
+    const [tableData, setTableData] = React.useState<MaintenanceWindow[]>([]);
+
+    // Modal Consts
+    const [showEditNew, setShowEditNew] = React.useState<boolean>(false);
+    const [showDeleteWarning, setShowDeleteWarning] = React.useState<boolean>(false);
+    const [activeWindow, setActiveWindow] = React.useState<MaintenanceWindow>(null);
+    const [reset, setReset] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        let handle = getMaintenanceWindows();
+        return () => {
+            if (handle != null && handle.abort != null)
+                handle.abort();
+        };
+    }, [ascending, sortKey, props.Meter, reset]);
+
+    function getMaintenanceWindows(): JQuery.jqXHR<MaintenanceWindow[]> {
+        setStatus('loading');
+        let handle = $.ajax({
+            type: 'POST',
+            url: `${homePath}api/OpenXDA/MaintenanceWindow/SearchableList`,
+            contentType: "application/json; charset=utf-8",
+            data: JSON.stringify({
+                Searches: [{ FieldName: 'MeterID', Operator: "=", SearchText: props.Meter.ID, Type: 'string', isPivotColumn: false }],
```

Comment:
This works Ok, But for future Reference there is a `ParentKey` attribute you can add to the model an then everything, including the `GET` without search will be restricted to a ParentID.